agent/platform.go

```diff
@@ -29,6 +30,11 @@ func NewPlatform(ctx context.Context, ignoreContainer *regexp.Regexp) (platform.
 	case platform.ECSAwsvpc:
 		return ecsawsvpc.NewECSAwsvpcPlatform(false, ignoreContainer)
 
+	case platform.ECSv3:
+		metadataURI := os.Getenv("ECS_CONTAINER_METADATA_URI")
```

Comment:
What happens if users mistakingly specify ECSv3 at unsupported environments like old container instances (or Fargate)?
Maybe it doesn't cause disaster 💥 with current implementation, but users will be happy if they can get feedback to know the container-agent is in unsupported environments.

Reply:
In the current implementation, the connection error to the endpoint is output in the log and the agent terminates.
Fix to give the user appropriate feedback.